From 4ae6f3dab170568760ed3f00edc73da90c65d13f Mon Sep 17 00:00:00 2001
From: Justus Klausecker <justus@klausecker.de>
Date: Sat, 13 Sep 2025 14:35:48 +0200
Subject: [PATCH 102/133] std.os.linux.ptrace: add PTRACE_EVENT_* and
 PTRACE_O_* values
X-Developer-Signature: v=1; a=openpgp-sha256; l=1576; i=xtex@astrafall.org;
 h=from:subject; bh=gXGvH8x23RJuKQNEQf4NM4BVqtvhNOJuz5Mx2vl0dvo=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78rijN5kc/JlQc5dr1Zb2Pn+dlCL7NY7z/CWiHfV2f
 fqXlCffMZGFQYyLwVJMkaXIsMGbVSedX3RZuSzMHFYmkCHSIg0MDAwMLAx8uYl5pUY6Rnqm2oZ6
 hoY6RjpGDFycAjDVvk0M/wuVW02kY98ZbkpauFlqxRzVs0aNtaXzcxfURxY9z5w/Zy3D/8SEC8z
 1BqtDvx50OW4bemfO47gtHjNqQgK5Xa9uTS9r5wcA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 1b4508cfb366be940ae39b458c632f6952116a65)
---
 lib/std/os/linux.zig | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/lib/std/os/linux.zig b/lib/std/os/linux.zig
index c2aa40f5520c..1b0b752f76fd 100644
--- a/lib/std/os/linux.zig
+++ b/lib/std/os/linux.zig
@@ -9648,6 +9648,33 @@ pub const PTRACE = struct {
     pub const SECCOMP_GET_FILTER = 0x420c;
     pub const SECCOMP_GET_METADATA = 0x420d;
     pub const GET_SYSCALL_INFO = 0x420e;
+
+    pub const EVENT = struct {
+        pub const FORK = 1;
+        pub const VFORK = 2;
+        pub const CLONE = 3;
+        pub const EXEC = 4;
+        pub const VFORK_DONE = 5;
+        pub const EXIT = 6;
+        pub const SECCOMP = 7;
+        pub const STOP = 128;
+    };
+
+    pub const O = struct {
+        pub const TRACESYSGOOD = 1;
+        pub const TRACEFORK = 1 << PTRACE.EVENT.FORK;
+        pub const TRACEVFORK = 1 << PTRACE.EVENT.VFORK;
+        pub const TRACECLONE = 1 << PTRACE.EVENT.CLONE;
+        pub const TRACEEXEC = 1 << PTRACE.EVENT.EXEC;
+        pub const TRACEVFORKDONE = 1 << PTRACE.EVENT.VFORK_DONE;
+        pub const TRACEEXIT = 1 << PTRACE.EVENT.EXIT;
+        pub const TRACESECCOMP = 1 << PTRACE.EVENT.SECCOMP;
+
+        pub const EXITKILL = 1 << 20;
+        pub const SUSPEND_SECCOMP = 1 << 21;
+
+        pub const MASK = 0x000000ff | PTRACE.O.EXITKILL | PTRACE.O.SUSPEND_SECCOMP;
+    };
 };
 
 /// For futex2_waitv and futex2_requeue. Arrays of `futex2_waitone` allow
-- 
2.52.0

